extensions/sql-database-projects/src/test/projectController.test.ts

```diff
@@ -678,6 +678,47 @@ describe('ProjectsController', function (): void {
 			should(projFileText).containEql('..\\someFolder\\outsideFolderTest.dacpac');
 		});
 	});
+
+	describe('AutoRest generation', function (): void {
+		it('Should create project from autorest-generated files', async function (): Promise<void> {
+			const parentFolder = await testUtils.generateTestFolderPath();
+			await testUtils.createDummyFileStructure();
+			const specName = 'DummySpec.yaml';
+			const newProjFolder = path.join(parentFolder, path.basename(specName, '.yaml'));
+
+			const projController = TypeMoq.Mock.ofType(ProjectsController);
+			projController.callBase = true;
+
+			projController.setup(x => x.selectAutorestSpecFile()).returns(async () => specName);
+			projController.setup(x => x.selectAutorestProjectLocation(TypeMoq.It.isAny())).returns(async () => {
+				await fs.mkdir(newProjFolder);
+
+				return {
+					newProjectFolder: newProjFolder,
+					outputFolder: parentFolder,
+					projectName: path.basename(specName, '.yaml')
+				};
+			});
+
+			let fileList: vscode.Uri[] = [];
+			projController.setup(x => x.generateAutorestFiles(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(async () => {
+				await testUtils.createDummyFileStructure(true, fileList, newProjFolder);
```

Comment:
I want to rework `createDummyFileStructure()` to be more expressive about what it has generated instead of just returning a flat list of folders + files.  Note to self for another PR.